packages/palette/src/utils/usePosition.ts

```diff
@@ -0,0 +1,373 @@
+/**
+ * Adapted from https://codesandbox.io/s/positioning-tooltip-rhplo
```

Comment:
I thought this code was very clear and brief. I added support for offset, deactivating, window resizing, and converted it to a hook.

Alternatively we could just use [@popperjs/core](https://popper.js.org/docs/v2/).